ci/script.sh

```diff
@@ -2,6 +2,14 @@
 
 set -e
 
+curl https://sh.rustup.rs -sSf | sh -s -- -y
+export PATH="$HOME/.cargo/bin:$PATH"
+
+eval `ssh-agent -s`
+ssh-add
+
+CARGO_NET_GIT_FETCH_WITH_CLI=true
```

Comment:
Workaround to rust ssh authentication... 
```
Caused by:
  failed to authenticate when downloading repository: ssh://git@github.com/rust-lang/crates.io-index

  * attempted ssh-agent authentication, but no usernames succeeded: `git`
 ```

https://sathia27.github.io/posts/2021/08/19/rust-cargo-resolve-authentication-issue.html
https://doc.rust-lang.org/nightly/cargo/appendix/git-authentication.html#git-authentication

Reply:
However `ci/script.sh` worked without any of this additional ssh config when I sshed into circleci?

Reply:
Getting the error once again on a new commit?